draft-ietf-ice-pac.xml

```diff
@@ -180,7 +180,20 @@
         equal to the agent's connectivity check transaction timeout.
       </t>
       <t>
-        This timeout value is chosen to roughly coincide with the maximum
+        Specifically, the ICE agent will start its timer once it expects to
+        receive ICE connectivity checks. This occurs when the agent has
```

Comment:
I think with your latest edge case of not handing out ICE candidates yourself this initial sentence could be misleading. In that scenario the ICE would never expect to receive ICE connectivity checks, as it only ever sends the connectivity checks itself.

Reply:
Good point, PTAL at the latest text.

Reply:
I am sorry, but I still think the text is confusing.

The text now says that the timer is started when the ICE agent has SENT values (userid, passwd), and RECEIVED some "indication" (e.g., the same values), but the ICE agent may not have received candidates yet.

Q: The text says nothing regarding whether the ICE agent as SENT candidates or not. If the agent has not sent or received any candidates, the procedure does not ensure that the timer "is started approximately when the remote peer starts doing connectivity checks". If no candidates have been exchanged, no endpoint will start doing connectivity checks. I believe Justin made the same comment (https://github.com/cdh4u/draft-ice-pac/pull/15#discussion_r302296222) a couple of days ago.

Q: The text says that the timer is started based on candidates that the remote peer later receives. How can the timer be started based on something that happens later?

Reply:
> I am sorry, but I still think the text is confusing.
> 
> The text now says that the timer is started when the ICE agent has SENT values (userid, passwd), and RECEIVED some "indication" (e.g., the same values), but the ICE agent may not have received candidates yet.

Correct.

> Q: The text says nothing regarding whether the ICE agent as SENT candidates or not. If the agent has not sent or received any candidates, the procedure does not ensure that the timer "is started approximately when the remote peer starts doing connectivity checks". If no candidates have been exchanged, no endpoint will start doing connectivity checks. I believe Justin made the same comment ([#15 (comment)](https://github.com/cdh4u/draft-ice-pac/pull/15#discussion_r302296222)) a couple of days ago.

Right, that's the whole point. We don't want to wait to send candidates because we still may need the timer even if we don't send candidates. I'm not sure I understand your concern here.

> Q: The text says that the timer is started based on candidates that the remote peer later receives. How can the timer be started based on something that happens later?

Yeah, I see your point here. I don't think we need to change the behavior but I will update the text.

Reply:
>> I am sorry, but I still think the text is confusing.
>> The text now says that the timer is started when the ICE agent 
>> has SENT values (userid, passwd), and RECEIVED some "indication" 
>> (e.g., the same values), but the ICE agent may not have received candidates yet.
>> 
>> Correct.
>> 
>> Q: The text says nothing regarding whether the ICE agent as SENT candidates 
>> or not. If the agent has not sent or received any candidates, the procedure does 
>> not ensure that the timer "is started approximately when the remote peer starts doing 
>> connectivity checks". If no candidates have been exchanged, no endpoint will start doing >>connectivity checks. I believe Justin made the same comment (#15 (comment)) a couple of 
>> days ago.
> 
> Right, that's the whole point. We don't want to wait to send candidates because we still may >need the timer even if we don't send candidates. I'm not sure I understand your concern here.

If we want to start the timer once only the ICE credentials (userid, pwd) have been exchanged, that is one thing.

But, then we can not say that it guarantees that the timer will be started approximately when the connectivity checks start, because the connectivity checks will not start when only the ICE credentials have been exchanged.

Maybe I don't understand the issue that triggered the change: You were earlier talking about "having received an IP address". I then asked whether an "IP address" is a candidate, and you said yes. So, how can you receive an "IP address" if you have not sent or received a candidate?

Reply:
I clarified the problem description in #12, PTAL. Generally though the point is to simplify the timer; waiting until a candidate is sent or received makes things more complicated and doesn't really provide much benefit, other than early detection of the zero-candidates-on-either-side case.

Reply:
> I clarified the problem description in #12, PTAL. Generally though the point is to simplify the 
> timer; waiting until a candidate is sent or received makes things more complicated and doesn't > really provide much benefit, other than early detection of the zero-candidates-on-either-side 
> case.

I have no problem to start the timer once the ICE credentials have been exchanged - even if no candidate has been sent or received.

My point is that, in that case, we can't say that it guarantees that the timer is started approximately when the connectivity checks start, because if there is no candidate that statement is not true. So, why don't we simply remove that part?

Reply:
OK, I agree that last sentence isn't required. Removed, PTAL.

Reply:
Please make it clear that no candidate needs to be received OR sent.

Something like:

"note that it is not required for the ICE agent to send or receive an ICE candidate before the timer is started."

Otherwise I am fine with the text.